ExtensionManager/ExtensionManager/ViewModels/ExtensionBrowserViewModel.cs

```diff
@@ -85,61 +59,39 @@ public ExtensionBrowserViewModel(ExtensionManagerSettings settings)
             PackageManager.FileSystem.Logger = PackageManager.Logger;
             repository.Logger = PackageManager.Logger;
 
-            var addExt = new AddExtensionCommand(ExtensionsDirectory, repository, this);
-            AddExtensionCommand = addExt;
-
-            var remExt = new RemoveExtensionCommand(ExtensionsDirectory, repository, this);
-            RemoveExtensionCommand = remExt;
-
-            var updateExt = new UpdateExtensionCommand(ExtensionsDirectory, repository, this);
-            UpdateExtensionCommand = updateExt;
-
-            PropertyChanged += (sender, args) =>
-            {
-                if (args.PropertyName != "Progress") return;
-                Debug.WriteLine("ShowProgress changing");
-                PropertyChanged?.Invoke(this, new PropertyChangedEventArgs("ShowProgress"));
-            };
-            
-            WorkStarted += (sender, e) => Progress = 1;
-            WorkComplete += (sender, e) => Progress = 0;
+            AddExtensionCommand = new AddExtensionCommand(this);
+            RemoveExtensionCommand = new RemoveExtensionCommand(this);
+            UpdateExtensionCommand = new UpdateExtensionCommand(this);
 
             RefreshPackageList();
         }
 
-        public void RaiseWorkStarted()
+        internal void RestartApplicationWithConfirm()
         {
-            Debug.WriteLine("Raising WorkStarted");
-            WorkStarted?.Invoke(this, EventArgs.Empty);
-        }
+            var restart = MessageBoxFactory.ShowYesNo("Requested operation has been completed. irAuthor must be restarted before your changes can take effect. Would you like to restart irAuthor now?", "Restart needed", MessageBoxFactoryImage.Question, ExtensionBrowserView);
 
-        public void RaiseWorkComplete()
-        {
-            Debug.WriteLine("Raising WorkComplete");
-            WorkComplete?.Invoke(this, EventArgs.Empty);
-        }
+            if (!restart) return;
 
-        internal void RestartApplicationWithConfirm()
-        {
-            var result = MessageBox.Show(ExtensionBrowserView,
-                       "Requested operation has been completed. IrAuthor must be restarted before your changes can take effect. Would you like to restart IrAuthor now?",
-                       "Restart needed", MessageBoxButton.YesNo);
+            // If a ruleapp is loaded, give them the opportunity to save it or cancel.
+            if (RuleApplicationService.RuleApplicationDef != null)
+            {
+                restart = RuleApplicationService.Close();
+            }
 
-            if (result != MessageBoxResult.Yes) return;
+            if (!restart) return;
 
-            System.Diagnostics.Process.Start(Application.ResourceAssembly.Location);
-            Application.Current.Shutdown();
+            Process.Start(Application.ResourceAssembly.Location);
```

Comment:
We're not currently attempting to re-launch irAuthor with any command-line parameters passed to the original invocation. Do you think it's necessary to do that?

Reply:
we can file an issue against this if needed. closing for now